pkg/kube/podtemplatespec/podspec_template_test.go

```diff
@@ -247,6 +247,140 @@ func TestMergeEnvironmentVariables(t *testing.T) {
 	assert.Equal(t, mergedContainer.Env[1].Value, "val2")
 }
 
+func TestMergeTolerations(t *testing.T) {
+	tests := []struct {
+		name                string
+		defaultTolerations  []corev1.Toleration
+		overrideTolerations []corev1.Toleration
+		expectedTolerations []corev1.Toleration
+	}{
+		{
+			// In case the calling code specifies default tolerations,
+			// they should be kept when there are no overrides.
+			name: "Overriding with nil tolerations",
+			defaultTolerations: []corev1.Toleration{
+				{
+					Key:      "key1",
+					Value:    "value1",
+					Operator: corev1.TolerationOpEqual,
+				},
+				{
+					Key:      "key1",
+					Value:    "value2",
+					Operator: corev1.TolerationOpExists,
+				},
+			},
+			overrideTolerations: nil,
+			expectedTolerations: []corev1.Toleration{
+				{
+					Key:      "key1",
+					Value:    "value1",
+					Operator: corev1.TolerationOpEqual,
+				},
+				{
+					Key:      "key1",
+					Value:    "value2",
+					Operator: corev1.TolerationOpExists,
+				},
+			},
+		},
+		{
+			// If the override is specifying an empty list of tolerations,
+			// they should replace default tolerations.
+			name: "Overriding with empty tolerations",
+			defaultTolerations: []corev1.Toleration{
+				{
+					Key:      "key1",
+					Value:    "value1",
+					Operator: corev1.TolerationOpEqual,
+				},
+			},
+			overrideTolerations: []corev1.Toleration{},
+			expectedTolerations: []corev1.Toleration{},
+		},
+		{
+			// Overriding toleration should replace a nil original toleration.
+			name:               "Overriding when default toleration is nil",
+			defaultTolerations: nil,
```

Comment:
Would it make sense to also test when defaultTolerations is an empty array?

Reply:
This is identical to the case when `defaultTolerations` is a non-empty array, since we discard everything there. Should we change the logic, that test might make sense.